ESTADISTICA

PRUEBA RESET DE RAMSEY:

La prueba RESET de Ramsey (Regression Specification Error Test) sirve para detectar errores de especificación en un modelo de regresión lineal, particularmente:

✅ Omisión de variables relevantes
✅ Mala forma funcional (el modelo debería incluir términos no lineales y no los incluiste)
✅ Interacciones no consideradas
✅ Variables mal transformadas

En otras palabras, la prueba analiza si el modelo está correctamente especificado o si le falta estructura.

¿Cómo funciona en términos simples?
1. Estimas tu modelo original por MCO.
2. Luego agregas al modelo potencias de los valores ajustados
3. Si esos términos adicionales son estadísticamente significativos → hay evidencia de que tu modelo está mal especificado.

📌 Hipótesis:
1. H0: El modelo esta bien especificado
2. H1: El modelo está mal especificado

Si se rechaza H0, significa que el modelo podria estar incompleto o mal planteado y que se deben revisar variables,
transformaciones o forma funcional.

La prueba reset me dice si el modelo lineal se quedó corto y si le falto algo


In [8]:
import statsmodels.api as sm

from statsmodels.stats.diagnostic import linear_reset
import pandas as pd

# Ejemplo de datos
data = {
    "y": [5, 7, 6, 9, 12, 15, 14, 18, 20, 21],
    "x": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
}

df = pd.DataFrame(data)

X = sm.add_constant(df["x"])
y = df["y"]

# Modelo MCO
model = sm.OLS(y, X).fit()
print(model.summary())

# Prueba RESET de Ramsey
reset_test = linear_reset(model, power=2, use_f=True) 
print("\nResultado RESET:")
print(reset_test)




                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.964
Model:                            OLS   Adj. R-squared:                  0.959
Method:                 Least Squares   F-statistic:                     211.6
Date:                Sat, 01 Nov 2025   Prob (F-statistic):           4.89e-07
Time:                        14:08:23   Log-Likelihood:                -14.767
No. Observations:                  10   AIC:                             33.53
Df Residuals:                       8   BIC:                             34.14
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          2.2667      0.809      2.801      0.0

c:\Users\Usuario\AppData\Local\Programs\Python\Python39\lib\site-packages\scipy\stats\_axis_nan_policy.py:531: UserWarning: kurtosistest only valid for n>=20 ... continuing anyway, n=10
  res = hypotest_fun_out(*samples, **kwds)


In [ ]:
# Prueba RESET de Ramsey
reset_test = linear_reset(model, power=2, use_f=True) 
print("\nResultado RESET:")
print(reset_test)


Resultado RESET:
<F test: F=0.4048265029272803, p=0.544841506223661, df_denom=7, df_num=1>


p-valor = 0.5448 > 0.05 → No rechazamos. No hay evidencia estadística de mala especificación del modelo.
Es decir, el modelo lineal parece adecuado y no faltan términos importantes

TEST DE TUKEY

¿Qué es el test de Tukey?
Es una prueba post-hoc para comparar medias de más de dos grupos de forma pareada (grupo 1 vs 2, 1 vs 3, 2 vs 3, etc.), controlando el error tipo I cuando se hacen muchas comparaciones.

ANOVA te dice "al menos dos grupos son distintos"
Tukey te dice "cuáles grupos son distintos"

Se usa cuando:
1. Hay 3 o más grupos
2. Hiciste un ANOVA significativo
3. Las varianzas son más o menos homogéneas

Si ANOVA no es significativo → Tukey no se realiza

Si el p-valor < 0.05, los grupos son significativamente distintos

PRUEBA DE WILCOXON-MANN-WHITNEY Y PRUEBA DE KRUSKALL-WALLIS



Ambas son pruebas no paramétricas, no asumen normalidad en los datos, pero se aplican en situaciones distintas.

1. Prueba de Wilcoxon–Mann–Whitney: Comparar dos grupos independientes para evaluar si provienen de la misma distribución (especialmente si tienen igual mediana).

2. Prueba de Kruskal–Wallis: Es un ANOVA no parametrico, Comparar tres o más grupos independientes para determinar si al menos uno difiere en su distribución o mediana. 

TEST DE SHAPIRO-WIK

Test de Shapiro–Wilk: Evaluar si una muestra proviene de una distribución normal

El valor W compara la dispersión observada de los datos con la esperada bajo una distribución normal. Cuanto más se aleja de 1, mayor es la desviación de la normalidad.


PRUEBA LEVENE

La prueba de Levene se utiliza en estadística para verificar la homogeneidad de varianzas entre dos o más grupos.

PCA

Principal Components Analisis